### PR TITLE
Addition of Sparsemax activation

### PR DESCRIPTION
**Pull request by old-school-kid**
PR for creating the sparsemax activation layer as requested in issue [#20554](https://github.com/keras-team/keras/issues/20554). 
[Original paper](https://arxiv.org/abs/1602.02068)

### CODE REVIEW

File: keras/src/activations/activations.py [hunk 1]
```diff
@@ -617,3 +617,31 @@ def log_softmax(x, axis=-1):
...
+
+
+@keras_export(["keras.activations.sparsemax"])
+def sparsemax(x, axis=-1):
+    """Sparsemax activation function.
+
+    For each batch $i$, and class $j$,
+    compute sparsemax activation function:
+
+    $$
```
Comment (fchollet) [inline keras/src/activations/activations.py:629]: Just use a regular Python code block with triple backticks, ```python

File: keras/src/activations/activations.py [hunk 2]
```diff
@@ -617,3 +617,31 @@ def log_softmax(x, axis=-1):
...
+    \mathrm{sparsemax}(x)[i, j] = \max(\mathrm{logits}[i, j] -
+    \tau(\mathrm{logits}[i, :]), 0).
+    $$
+
+    Args:
+        logits: A `Tensor`.
```
Comment (fchollet) [inline keras/src/activations/activations.py:635]: "A tensor" -- it's not a specific class

File: keras/src/ops/nn.py [hunk 3]
```diff
@@ -951,6 +951,53 @@ def log_softmax(x, axis=-1):
...
+
+    def call(self, x):
+        return backend.nn.sparsemax(x, axis=self.axis)
+
+    def compute_output_spec(self, x):
+        return KerasTensor(x.shape, dtype=x.dtype)
+
+
+@keras_export(["keras.ops.sparsemax", "keras.ops.nn.sparsemax"])
+def sparsemax(x, axis=-1):
```
Comment (fchollet) [inline keras/src/ops/nn.py:967]: Same comments for this docstring

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras/src/ops/nn.py [hunk 4]
```diff
@@ -951,6 +951,48 @@ def log_softmax(x, axis=-1):
...
+    """Sparsemax activation function.
+
+    For each batch `i`, and class `j`,
+    sparsemax activation function is defined as:
+
+    `sparsemax(x)[i, j] = max(logits[i, j] - τ(logits[i, :]), 0).`
+
+    Args:
+        logits: Input tensor.
```
Comment (fchollet) [inline keras/src/ops/nn.py:976]: It's x

File: keras/src/activations/activations.py [hunk 5]
```diff
@@ -617,3 +617,28 @@ def log_softmax(x, axis=-1):
...
+def sparsemax(x, axis=-1):
+    """Sparsemax activation function.
+
+    For each batch `i`, and class `j`,
+    sparsemax activation function is defined as:
+
+    `sparsemax(x)[i, j] = max(logits[i, j] - τ(logits[i, :]), 0).`
+
+    Args:
+        logits: Input tensor.
```
Comment (fchollet) [inline keras/src/activations/activations.py:632]: Same here that's x

File: keras/src/activations/activations.py [hunk 6]
```diff
@@ -617,3 +617,28 @@ def log_softmax(x, axis=-1):
...
+
+
+@keras_export(["keras.activations.sparsemax"])
+def sparsemax(x, axis=-1):
+    """Sparsemax activation function.
+
+    For each batch `i`, and class `j`,
+    sparsemax activation function is defined as:
+
+    `sparsemax(x)[i, j] = max(logits[i, j] - τ(logits[i, :]), 0).`
```
Comment (fchollet) [inline keras/src/activations/activations.py:629]: logits -> x

**Review by fchollet — APPROVED** [review]
LGTM, thank you for the contribution! 👍
